packages/grafana-ui/src/components/NodeGraph/NodeGraph.tsx

```diff
@@ -0,0 +1,243 @@
+  // sure they are visible on top of everything else
+  const [nodeHover, setNodeHover] = useState<string | undefined>(undefined);
+  const clearNodeHover = useCallback(() => setNodeHover(undefined), [setNodeHover]);
+  const [edgeHover, setEdgeHover] = useState<string | undefined>(undefined);
+  const clearEdgeHover = useCallback(() => setEdgeHover(undefined), [setEdgeHover]);
+
+  // TODO we should be able to allow multiple dataframes for both edges and nodes, could be issue with node ids which in
+  //  that case should be unique or figure a way to link edges and nodes dataframes together.
+  const processed = useMemo(() => processNodes(nodesDataFrames[0], edgesDataFrames[0]), [
+    nodesDataFrames[0],
+    edgesDataFrames[0],
+  ]);
+
+  const { nodes: rawNodes, edges: rawEdges } = useNodeLimit(processed.nodes, processed.edges, nodeCountLimit);
+  const hiddenNodesCount = processed.nodes.length - rawNodes.length;
+
+  const { nodes, edges, bounds } = useLayout(rawNodes, rawEdges, config);
+  const { panRef, zoomRef, onStepUp, onStepDown, isPanning, position, scale } = usePanAndZoom(bounds);
+  const { onEdgeOpen, onNodeOpen, MenuComponent } = useContextMenu(getLinks, nodesDataFrames[0], edgesDataFrames[0]);
+  const styles = getStyles(useTheme());
+
+  return (
+    <div
+      ref={r => {
+        measureRef(r);
+        (zoomRef as MutableRefObject<HTMLElement | null>).current = r;
+      }}
+      className={styles.wrapper}
+    >
+      <svg
+        ref={panRef}
+        viewBox={`${-(width / 2)} ${-(height / 2)} ${width} ${height}`}
+        className={cx(styles.svg, isPanning && styles.svgPanning)}
+      >
+        <g
+          className={styles.mainGroup}
+          style={{ transform: `scale(${scale}) translate(${Math.floor(position.x)}px, ${Math.floor(position.y)}px)` }}
+        >
+          <EdgeArrowMarker />
+          <Edges
+            edges={edges}
+            nodeHoveringId={nodeHover}
+            edgeHoveringId={edgeHover}
+            onClick={onEdgeOpen}
+            onMouseEnter={setEdgeHover}
+            onMouseLeave={clearEdgeHover}
+          />
+          <Nodes
+            nodes={nodes}
+            onMouseEnter={setNodeHover}
+            onMouseLeave={clearNodeHover}
+            onClick={onNodeOpen}
+            hoveringId={nodeHover}
+          />
+          {/*We split the labels from edges so that they are shown on top of everything else*/}
+          <EdgeLabels edges={edges} nodeHoveringId={nodeHover} edgeHoveringId={edgeHover} />
+        </g>
+      </svg>
+
+      <div className={styles.viewControls}>
+        <ViewControls<Config>
+          config={config}
+          onConfigChange={setConfig}
+          onMinus={onStepDown}
+          onPlus={onStepUp}
+          scale={scale}
+        />
+      </div>
+
+      {hiddenNodesCount > 0 && (
+        <div className={styles.alert} aria-label={'Nodes hidden warning'}>
```

Comment:
I see that you are doing this because we "miss" a warning variant for the alert component (meaning we have the variant but it's identical to the error one), I think there was some discussion on reintroducing it (with the orange background), so maybe 

```ts
<div className={styles.alertContainer}>
  <Alert title="" severity="warning">
    {hiddenNodesCount} nodes are hidden for performance reasons.
  </Alert>
</div>
```

and, instead of https://github.com/grafana/grafana/pull/29706/files#diff-ffd3d08d366e741396b93d7037a7145086a2ccfa69e4c86f502e8dcde02ed879R51 :

```ts
  alertContainer: css`
    position: absolute;
    top: 0;
    right: 0;
  `,
```

what do you think?

also, the reason is that the text is way too small on high-res screens:
![Screenshot 2021-01-19 at 09 55 40](https://user-images.githubusercontent.com/1170767/105019957-9c331e00-5a3e-11eb-95bc-37dc65e9b287.png)
![Screenshot 2021-01-19 at 10 10 28](https://user-images.githubusercontent.com/1170767/105019959-9ccbb480-5a3e-11eb-8f43-97453ca8b51a.png)

Reply:
So do you propose to use the red warning for the time being? I wanted to make this not too flashy, it's our problem and I don't wan't people to confuse it with any other error. Also made it smaller because of that don't wan't it to take more space than needed.

Reply:
I think we should make the `severity="warning"` different from the `severity="error"` alert, but i'd rather be consistent with Grafana general look and feel than implementing custom UI components/patterns that are different from what is found everywhere else. So maybe I would push forward for a change in @grafana/ui (since we actually need it) than implementing a custom alert notification.

Reply:
Yeah I agree but for the sake of this particular PR let's keep it as it is and we can change once the alert styles change